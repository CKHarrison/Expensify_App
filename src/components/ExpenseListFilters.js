import React from 'react';
import { connect } from 'react-redux';
import {setTextFilter} from '../actions/filters';

const ExpenseListFilters = (props) => (
  <div>
    <input type="text" value={props.filters.text} onChange={(e) => {
      // This will track the key stroke and modify store to reflect that
      props.dispatch(setTextFilter(e.target.value));
    }}/>
    <select value={props.filters.text}>
      <option value="date" onChange={(e) => {
        // This will track the key stroke and modify store to reflect that
        props.dispatch(setTextFilter(e.target.value));
      }}>Date</option>
      <option value="amount">Amount</option>
    </select>
  </div>
);

const mapStateToProps = (state) => {
  return {
    filters: state.filters
  };
};

export default connect(mapStateToProps)(ExpenseListFilters);