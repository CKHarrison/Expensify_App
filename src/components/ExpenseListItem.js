import React from 'react';
import ExpenseList from './ExpenseList';
import {connect} from 'react-redux';
import {removeExpense} from '../actions/expenses';

const ExpenseListItem = ({dispatch, description, amount, createdAt, id}) => (
  <div>
    <h3>{description}</h3>
    <p>Amount: {amount} - Created At: {createdAt}</p>
    <button onClick={() => {
      dispatch(removeExpense({id}));
    }}>Remove</button>
  </div>
);


export default connect()(ExpenseListItem);