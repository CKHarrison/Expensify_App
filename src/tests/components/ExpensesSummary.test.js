import React from 'react';
import { shallow } from 'enzyme';
import expenses from '../fixtures/expenses';
import {ExpensesSummary} from '../../components/ExpensesSummary';


test('should correctly render ExpenseSummary with one expenses', () =>{
  const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={400}/>);
  expect(wrapper).toMatchSnapshot();
});

test('should correctly render ExpenseSummary with one expenses', () =>{
  const wrapper = shallow(<ExpensesSummary expenseCount={23} expensesTotal={4004564879321}/>);
  expect(wrapper).toMatchSnapshot();
});