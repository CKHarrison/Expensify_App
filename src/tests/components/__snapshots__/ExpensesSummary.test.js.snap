// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render ExpenseSummary with one expenses 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totaling 
    $4.00
  </h1>
</div>
`;

exports[`should correctly render ExpenseSummary with one expenses 2`] = `
<div>
  <h1>
    Viewing 
    23
     
    expenses
     totaling 
    $40,045,648,793.21
  </h1>
</div>
`;
